app/lib/get-user-ribbon-data.js

```diff
@@ -0,0 +1,43 @@
+import getAllProjectPreferences from './get-all-project-preferences';
+import getColorFromString from './get-color-from-string';
+
+function getUserRibbonData(user) {
+  return getAllProjectPreferences(user).then((prefsResources) => {
+    const awaitProjects = Promise.all(prefsResources.map((prefsResource) => {
+      return prefsResource.get('project').catch(() => {
+        return null;
+      });
+    }));
+
+    const awaitOwners = awaitProjects.then((projects) => {
+      return Promise.all(projects.map((project) => {
+        if (project === null) {
+          return null;
+        } else {
+          return project.get('owner').catch(() => {
```

Comment:
All you need the owner for later on is the `display_name`. This is already given to you in a project's `links` section.